packages/graphql/src/translate/where/property-operations/create-comparison-operation.ts

```diff
@@ -0,0 +1,120 @@
+import { createPointComparisonOperation } from "./create-point-comparison-operation";
+
+/** Translates an atomic comparison operation (e.g. "this0 <= $param0") */
+export function createComparisonOperation({
+    operator,
+    propertyRefOrCoalesce,
+    param,
+    durationField,
+    pointField,
+}: {
+    operator: string | undefined;
+    propertyRefOrCoalesce: CypherBuilder.PropertyRef | CypherBuilder.Function;
+    param: CypherBuilder.Param;
+    durationField: PrimitiveField | undefined;
+    pointField: PointField | undefined;
+}): CypherBuilder.ComparisonOp {
+    if (pointField) {
+        return createPointComparisonOperation({
+            operator,
+            propertyRefOrCoalesce,
+            param,
+            pointField,
+        });
+    }
+
+    // Comparison operations requires adding dates to durations
+    // See https://neo4j.com/developer/cypher/dates-datetimes-durations/#comparing-filtering-values
+    if (durationField && operator) {
+        return createDurationOperation({ operator, property: propertyRefOrCoalesce, param });
+    }
+
+    return createBaseOperation({
+        operator: operator || "EQ",
+        property: propertyRefOrCoalesce,
+        param,
+    });
+}
+
+function createDurationOperation({
+    operator,
+    property,
+    param,
+}: {
+    operator: string;
+    property: CypherBuilder.Expr;
+    param: CypherBuilder.Expr;
+}) {
+    const variable = CypherBuilder.plus(CypherBuilder.datetime(), param);
+    const propertyRef = CypherBuilder.plus(CypherBuilder.datetime(), property);
+
+    return createBaseOperation({
+        operator,
+        property: propertyRef,
+        param: variable,
+    });
+}
+
+function createBaseOperation({
+    operator,
+    property,
+    param,
+}: {
+    operator: string;
+    property: CypherBuilder.Expr;
+    param: CypherBuilder.Expr;
+}): CypherBuilder.ComparisonOp {
+    switch (operator) {
+        case "LT":
+            return CypherBuilder.lt(property, param);
+        case "LTE":
+            return CypherBuilder.lte(property, param);
+        case "GT":
+            return CypherBuilder.gt(property, param);
+        case "GTE":
+            return CypherBuilder.gte(property, param);
+        case "ENDS_WITH":
+        case "NOT_ENDS_WITH":
+            return CypherBuilder.endsWith(property, param);
+        case "STARTS_WITH":
+        case "NOT_STARTS_WITH":
+            return CypherBuilder.startsWith(property, param);
+        case "MATCHES":
+            return CypherBuilder.matches(property, param);
+        case "CONTAINS":
+        case "NOT_CONTAINS":
+            return CypherBuilder.contains(property, param);
+        case "IN":
+        case "NOT_IN":
+            return CypherBuilder.in(property, param);
+        case "INCLUDES":
+        case "NOT_INCLUDES":
+            return CypherBuilder.in(param, property);
+        case "EQ":
+        case "NOT":
+            return CypherBuilder.eq(property, param);
```

Comment:
**question:** This is interesting, we're moving away from `x != y` to `NOT x = y`?

Reply:
I may be wrong, but I think we have never had != for this use case https://github.com/neo4j/graphql/blob/dev/packages/graphql/tests/tck/tck-test-files/advanced-filtering.test.ts#L121